### PR TITLE
Reenable precompiled headers in AspNetCore project

### PR DESCRIPTION
**Pull request by pakrym**
We need precomp.c because it causes precompiled header generation.

Additionally, we need PrecompiledHeader property to be set to Create for it.

And `#include "precomp.h"` should be the first line of every cpp file, putting it into .h was my mistake.

### CODE REVIEW

**Comment by pakrym** [comment]
CI Failure is unrelated https://github.com/aspnet/IISIntegration/issues/939

**Comment by jkotalik** [comment]
We have a mixed model of precompiled headers throughout our projects. @pan-wang you mentioned in [this PR](https://github.com/aspnet/IISIntegration/pull/916) that you wanted us to use our own precompiled headers. By doing this change, we are going back to VS style of precompiled headers. 

I currently don't see and precompiled headers at all on a default build of our stuff, so I think we have to go with the VS style. @pakrym you probably should change all other projects to use VS precompiled headers.

**Comment by pakrym** [comment]
Is there another way of doing precompiled headers? It's a compiler feature.

**Comment by pakrym** [comment]
Ping @pan-wang  @jkotalik

File: src/AspNetCoreModuleV2/AspNetCore/src/precomp.cpp [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// Copyright (c) .NET Foundation. All rights reserved.
```
Comment (jkotalik) [inline src/AspNetCoreModuleV2/AspNetCore/src/precomp.cpp]: I think you need to include precomp.h in this file.

File: src/AspNetCoreModuleV2/CommonLib/CommonLib.vcxproj [hunk 2]
```diff
@@ -100,6 +100,7 @@
       <AdditionalIncludeDirectories>..\iislib;</AdditionalIncludeDirectories>
       <MultiProcessorCompilation>true</MultiProcessorCompilation>
       <LanguageStandard>stdcpp17</LanguageStandard>
+      <ForcedIncludeFiles>stdafx.h</ForcedIncludeFiles>
```
Comment (jkotalik) [inline src/AspNetCoreModuleV2/CommonLib/CommonLib.vcxproj]: Why are you force including these?

File: src/AspNetCoreModuleV2/CommonLib/SRWExclusiveLock.h [hunk 3]
```diff
@@ -3,6 +3,8 @@
 
 #pragma once
 
+#include <synchapi.h>
```
Comment (jkotalik) [inline src/AspNetCoreModuleV2/CommonLib/SRWExclusiveLock.h]: ?

**Comment by jkotalik** [comment]
Do the RequestHandler projects use the same style of precompiled headers?

File: src/AspNetCoreModuleV2/CommonLib/CommonLib.vcxproj (hunk 2 above)
Comment (pakrym) [reply src/AspNetCoreModuleV2/CommonLib/CommonLib.vcxproj]: So I don't have to include in every cpp file.

File: src/AspNetCoreModuleV2/AspNetCore/src/precomp.cpp (hunk 1 above)
Comment (pakrym) [reply src/AspNetCoreModuleV2/AspNetCore/src/precomp.cpp]: It is included by `ForcedIncludeFiles`.

File: src/AspNetCoreModuleV2/CommonLib/SRWExclusiveLock.h (hunk 3 above)
Comment (pakrym) [reply src/AspNetCoreModuleV2/CommonLib/SRWExclusiveLock.h]: `SRWLOCK``

**Comment by pakrym** [comment]
> Do the RequestHandler projects use the same style of precompiled headers?

Not yet. I'll do it, just not in this PR.

**Comment by jkotalik** [comment]
PR is breaking things in dev.

**Pull request merged**